lua/entities/gmod_wire_expression2/core/egpfunctions.lua

```diff
@@ -817,16 +821,15 @@ end
 __e2setcost(20)
 e2function vector wirelink:egpGlobalPos( number index )
 	local hasvertices, posang = EGP:GetGlobalPos( this, index )
-	if (!hasvertices) then
-		return Vector( posang.x, posang.y, posang.angle )
+	if hasvertices then
```

Comment:
Do polygons not have a position you can translate them with? If they do, I'm not sure why we're returning a centroid position instead of the translation pos.

egpGetCenter (if that exists) would be more appropriate for getting the centroid pos

Reply:
Polygons do not have any position, they're directly from vertices. EGP objects are by default drawn from the center already. The only other position that could make sense is the first vertex. Maybe that could be something to add later.

Reply:
Ok, nevermind then.